src/gg.regression.unity.bots/Runtime/Scripts/RGServiceManager.cs

```diff
@@ -54,49 +54,29 @@ public bool IsAuthed()
             return rgAuthToken != null;
         }
 
-        private bool HasConfiguredTokenOrCredentials()
-        {
-            RGUserSettings rgSettings = RGUserSettings.GetOrCreateUserSettings();
-            string email = rgSettings.GetEmail();
-            string password = rgSettings.GetPassword();
-            var apiKey = RGEnvConfigs.ReadAPIKey();
-            if ((apiKey != null && apiKey.Trim() != "") || (!string.IsNullOrEmpty(email) && !string.IsNullOrEmpty(password)))
-            {
-                return true;
-            }
-            return false;
-        }
-
-        public async Task<bool> TryAuth()
+        public bool LoadAuth()
         {
             try
             {
                 // If an API key was given, just return and set that
                 var apiKey = RGEnvConfigs.ReadAPIKey();
                 if (apiKey != null && apiKey.Trim() != "")
                 {
-                    RGDebug.LogDebug("Using API Key from env var rather than username/password for auth");
+                    RGDebug.LogDebug("Using API Key from env var rather than from built in resources");
                     rgAuthToken = apiKey.Trim();
                 }
                 else
                 {
-                    RGDebug.LogDebug("Using username/password rather than env var for auth");
-                    RGUserSettings rgSettings = RGUserSettings.GetOrCreateUserSettings();
-                    string email = rgSettings.GetEmail();
-                    string password = rgSettings.GetPassword();
+                    RGSettings settings = RGSettings.GetOrCreateSettings();
+                    RGDebug.LogDebug("Using API Key from RGSettings resource");
 
-                    if (!string.IsNullOrEmpty(email) && !string.IsNullOrEmpty(password))
+                    if (!string.IsNullOrEmpty(settings.GetApiKey()))
                     {
-                        await Auth(
-                            email: email,
-                            password: password,
-                            onSuccess: s => { rgAuthToken = s; },
-                            onFailure: f => { }
-                        );
+                        rgAuthToken = settings.GetApiKey();
```

Comment:
this calls GetApiKey 2x, should only call it once

Reply:
Good catch, will fix.

Reply:
this is unresolved